### PR TITLE
Transform classic loops to range-based for loops in module search

### PR DESCRIPTION
**Pull request by SunBlack**
Changes are based on the result of run-clang-tidy -header-filter='.*' -checks='-*,modernize-loop-convert' -fix

### CODE REVIEW

File: search/include/pcl/search/impl/flann_search.hpp [hunk 1]
```diff
@@ -346,11 +345,11 @@ pcl::search::FlannSearch<PointT, FlannDistance>::radiusSearch (
   }
   if (!identity_mapping_)
   {
-    for (size_t j = 0; j < k_indices.size (); ++j )
+    for (auto &k_index : k_indices)
     {
-      for (size_t i = 0; i < k_indices[j].size (); ++i)
+      for (size_t i = 0; i < k_index.size (); ++i)
```
Comment (taketwo) [inline search/include/pcl/search/impl/flann_search.hpp]: What about `for (auto &neighbor_index : k_index)` here?

Comment (SunBlack) [reply search/include/pcl/search/impl/flann_search.hpp]: Well, seems clang-tidy don't like nested loops :(. Adjusted.

Comment (SergioRAgostinho) [reply search/include/pcl/search/impl/flann_search.hpp]: I don't think this was the adjustment @taketwo had in mind. I think he (and me as well) were expecting this line to changed to 
```cpp
for (auto &neighbor_index : k_index)
```
just like it happened in line 219

Comment (SunBlack) [reply search/include/pcl/search/impl/flann_search.hpp]: Ahh well, didn't saw this line occurred twice.

**Pull request merged**